intersect.go

```diff
@@ -95,6 +95,40 @@ func Difference[T comparable](list1 []T, list2 []T) ([]T, []T) {
 	return left, right
 }
 
+// Except returns distinct values from list1 except if that value is also in list2.
+func Except[T comparable](list1 []T, list2 []T) []T {
+	// bool is 1byte
+	// we are tracking what we want to _keep_
+	// with an initial value of false
+	// since we want to return distinct values only
+	// once we've seen the value, we'll change it to true
+	// and skip over future occurrences
+	keepSeen := make(map[T]bool)
```

Comment:
Value will never be `true`

can be changed by: `keepSeen := make(map[T]struct{})`

Reply:
done